src/Parser/Warlock/Demonology/Modules/Features/DEUptimePet.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import {formatPercentage} from 'common/format';
+import StatisticBox from 'Main/StatisticBox';
+
+import SPELLS from 'common/SPELLS';
+
+
+const DE_DURATION = 12;
+const MILLISECONDS = 1000;
+
+class DEUptimePet extends Analyzer{
```

Comment:
Are you manually tracking the buff for any particular reason? We provide several utilities for buff that could make this easier and more accurate, and I think this could be solved with `this.combatants.selected.getBuffUptime(SPELLS.DEMONIC_EMPOWERMENT.id)` instead?

Reply:
I ran into issues with examples where the player would die while their pet was buffed with demonic empowerment. WCL itself seems to assume that the buff on the pet just lasts the rest of the fight when this happens. After some brief discussion on discord, I decided to go with this manual tracking to get around the player death issue.